packages/dds/merge-tree/src/test/mergeTreeOperationRunner.ts

```diff
@@ -44,9 +44,13 @@ export const insertAtRefPos: TestOperation =
         if(segs.length > 0) {
             const text = client.longClientId!.repeat(random.integer(1, 3)(mt));
             const seg = random.pick(mt,segs);
-            return client.insertAtReferencePositionLocal(
-                new LocalReference(client, seg, random.integer(0, seg.cachedLength - 1)(mt)),
-                TextSegment.make(text));
+            const lref = new LocalReference(
+                client, seg, random.integer(0, seg.cachedLength - 1)(mt),
+                random.pick(mt,[ReferenceType.Simple, ReferenceType.SlideOnRemove, ReferenceType.Transient]));
+            if(lref.refType !== ReferenceType.Transient) {
```

Comment:
What makes Transient references special here? (in general docs on all of the ReferenceTypes would be great :))

Reply:
This is all legacy code with no documentation, so i don't have full clarity. They are also reused between a number of features, markers, local references, and shared intervals. Some types make some sense for some scenarios, but not others. It's an area that needs quite a bit of clean up.

Reply:
Within merge tree itself, slide on remove has special handling. when the segment is removed only local references with that type survive.